golink.go

```diff
@@ -286,6 +347,25 @@ func deleteLinkStats(link *Link) {
 	db.DeleteStats(link.Short)
 }
 
+// redirectHandler returns the http.Handler for serving all plaintext HTTP
+// requests. It redirects all requests to the HTTPs version of the same URL.
+func redirectHandler(hostname string) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		path := r.URL.Path
+		newUrl := fmt.Sprintf("https://%s%s", hostname, path)
+		http.Redirect(w, r, newUrl, http.StatusMovedPermanently)
```

Comment:
I'm not entirely sure this works, I vaguely recall something about `go/foo` not working once you start doing redirects.